src/kaleidoscope_internal/shortname.h

```diff
@@ -0,0 +1,34 @@
+/* kaleidoscope_internal::shortname -- HID short name override helper
+ * Copyright (C) 2019  Keyboard.io, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License as published by the Free Software
+ * Foundation, version 3.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+/*
+ * We want to be able to override the short name of the device. For that, we
+ * currently need to override `HID_::getShortName`, but due to link ordering, we
+ * need to do that in the user sketch. For this reason, `_INIT_HID_GETSHORTNAME`
+ * gets called from the `KEYMAPS` macro.
+ *
+ * TODO(anyone): Once we have a better way to override the short name, remove
+ * this workaround.
+ */
+
+#define _INIT_HID_GETSHORTNAME                       __NL__ \
+  uint8_t HID_::getShortName(char *name) {           __NL__ \
+    const char *n = Runtime.device().getShortName(); __NL__ \
```

Comment:
Although marked as `constexpr`, all device's `short_name` arrays are used at compile time, which will put them into SRAM. What about making them PROGMEM arrays and using `memcpy_P` instead. This would be similar to the keymap that is also declared `constexpr PROGMEM`.

Reply:
The problem with putting it into PROGMEM is that in that case, I need to add them not only to the header files, but to the cpp files too. Doable, not even hard, but it means another piece of boilerplate to add to every device.